packages/x-components/src/x-modules/scroll/store/utils.ts

```diff
@@ -0,0 +1,25 @@
+import Vue from 'vue';
+import { Dictionary } from '../../../utils/types';
+import { ScrollComponentState } from './types';
+
+/**
+ * Initialises an {@link ScrollComponentState} object if it has not already been created.
+ *
+ * @param state - The state shere the new data should be added.
+ * @param id - The identifier for the {@link ScrollComponentState}.
+ *
+ * @internal
+ */
+export function initScrollComponentState(
+  state: Dictionary<ScrollComponentState>,
+  id: string
+): void {
+  if (!state[id]) {
+    Vue.set<ScrollComponentState>(state, id, {
```

Comment:
I'd change this param name to `componentsState` to avoid confusions with the "whole" `store` `state` of the module.

```suggestion
 * Initialises an {@link ScrollComponentState} object if it has not already been created.
 *
 * @param componentsState - The componentsState where the new data should be added.
 * @param id - The identifier for the {@link ScrollComponentState}.
 *
 * @internal
 */
export function initScrollComponentState(
  componentsState: Dictionary<ScrollComponentState>,
  id: string
): void {
  if (!componentsState[id]) {
    Vue.set<ScrollComponentState>(componentsState, id, {
```